zebra/if_netlink.c

```diff
@@ -1964,26 +1888,38 @@ int netlink_vlan_change(struct nlmsghdr *h, ns_id_t ns_id, int startup)
 		if (!vtb[BRIDGE_VLANDB_ENTRY_STATE])
 			continue;
 
+		count++;
+		vlan_array =
+			XREALLOC(MTYPE_TMP, vlan_array,
+				 sizeof(struct zebra_vxlan_vlan_array) +
+					 count * sizeof(struct zebra_vxlan_vlan));
+
+		memset(&vlan_array->vlans[count - 1], 0,
+		       sizeof(struct zebra_vxlan_vlan));
+
 		state = *(uint8_t *)RTA_DATA(vtb[BRIDGE_VLANDB_ENTRY_STATE]);
 
 		if (vtb[BRIDGE_VLANDB_ENTRY_RANGE])
 			vrange = *(uint32_t *)RTA_DATA(
 				vtb[BRIDGE_VLANDB_ENTRY_RANGE]);
 
-		if (IS_ZEBRA_DEBUG_KERNEL || IS_ZEBRA_DEBUG_VXLAN) {
-			if (vrange)
-				zlog_debug("VLANDB_ENTRY: VID (%u-%u) state=%s",
-					   vinfo->vid, vrange,
-					   port_state2str(state));
-			else
-				zlog_debug("VLANDB_ENTRY: VID (%u) state=%s",
-					   vinfo->vid, port_state2str(state));
-		}
-
-		vlan_id_range_state_change(
-			ifp, vinfo->vid, (vrange ? vrange : vinfo->vid), state);
+		vlan_array->vlans[count - 1].state = state;
+		vlan_array->vlans[count - 1].vinfo = vinfo;
```

Comment:
I don't follow the use of "vinfo" here - that's a pointer, into the netlink message? that seems ... like a tough thing to capture and use later... maybe we should just capture the "vid" value directly, since that's what's being used?

Reply:
Mark is right, the vinfo should be copied instead of using the pointer into the netlink message.  This would be a recipe for some serious fun later on.

Reply:
Ack, makes sense to use vid directly..